Migrate rustc_mir_build diagnostics

mejrs opened the pull request:
Rebases https://github.com/rust-lang/rust/pull/100854

~~The remaining issue is how to better resolve https://github.com/rust-lang/rust/commit/72bea68af4ee2a41c44998916f6a789163f12e7d~~

~~The diagnostic macros seems to generate a broken diagnostic, and I couldn't figure out how to manually format the fluent message, so I hardcoded the format string for now. I'd like pointers to a better fix for this.~~

Also, I'm not 100% sure I didn't mess up a rebase somewhere 🙂 

r? @davidtwco

lzcunt committed:
Migrate "function cannot return without recursing" diagnostic

lzcunt committed:
Migrate "unsafe_op_in_unsafe_fn" lints

lzcunt committed:
Migrate "requires unsafe" diagnostics

lzcunt committed:
Migrate "unused unsafe" lint

lzcunt committed:
Migrate "non-exhaustive patterns: type is non-empty" diagnostic

lzcunt committed:
Migrate pattern inlining error diagnostics

lzcunt committed:
Migrate unreachable pattern diagnostic

lzcunt committed:
Migrate "constant pattern depends on generic parameter" diagnostic

lzcunt committed:
Migrate "could not evaluate const pattern" diagnostic

lzcunt committed:
Migrate lower range bound diagnostics

lzcunt committed:
Migrate leading/trailing irrefutable let pattern diagnostics

lzcunt committed:
Migrate pattern bindings with variant name lint

lzcunt committed:
Migrate irrefutable let pattern diagnostics

lzcunt committed:
Migrate borrow of moved value diagnostic

lzcunt committed:
Migrate multiple mut borrows diagnostic

bors commented:
:umbrella: The latest upstream changes (presumably #104418) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

davidtwco commented:
> The remaining issue is how to better resolve https://github.com/rust-lang/rust/commit/72bea68af4ee2a41c44998916f6a789163f12e7d
>
> The diagnostic macros seems to generate a broken diagnostic, and I couldn't figure out how to manually format the fluent message, so I hardcoded the format string for now. I'd like pointers to a better fix for this.

You should be able to fix this by using `#[subdiagnostic(eager)]` where this subdiagnostic is used.

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3470055676/jobs/5797782876) [(plain)](https://github.com/rust-lang/rust/commit/139bc67ef0864bc0a75a2a4105afb3c8b70cfd4b/checks/9497982479/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

```

</details>

mejrs commented:
> You should be able to fix this by using `#[subdiagnostic(eager)]` where this subdiagnostic is used.

Ah that's good to know. I've fixed that and rebased.

davidtwco commented on compiler/rustc_mir_build/src/thir/pattern/check_match.rs:
```diff
@@ -760,15 +676,17 @@ fn non_exhaustive_match<'p, 'tcx>(
+        });
+        return;
     } else {
+        // FIXME: migration of this diagnostic will require list support
```
You can try using the list support added in #104047, maybe better as a follow-up so we can land this though.

bors commented:
:pushpin: Commit 23546977081c6f3e3fc35afdc60f4c5a6b32e233 has been approved by `davidtwco`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=davidtwco 23546977081c6f3e3fc35afdc60f4c5a6b32e233 -->
<!-- homu: {"type":"Approved","sha":"23546977081c6f3e3fc35afdc60f4c5a6b32e233","approver":"davidtwco","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

Manishearth commented:
@bors r-

https://github.com/rust-lang/rust/pull/104790#issuecomment-1325615436

Commit:
Resolve various merge conflicts

mejrs commented:
Rebased

@rustbot ready

jyn514 commented:
@bors r=davidtwco

bors commented:
:pushpin: Commit 1d324dd126cd575e4887ac86e9c97e76908df123 has been approved by `davidtwco`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=davidtwco 1d324dd126cd575e4887ac86e9c97e76908df123 -->
<!-- homu: {"type":"Approved","sha":"1d324dd126cd575e4887ac86e9c97e76908df123","approver":"davidtwco","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 1d324dd126cd575e4887ac86e9c97e76908df123 with merge 3ece57ecf4c9f308fe935a733028dd53e5b2c69c...
<!-- homu: {"type":"BuildStarted","head_sha":"1d324dd126cd575e4887ac86e9c97e76908df123","merge_sha":"3ece57ecf4c9f308fe935a733028dd53e5b2c69c"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3672847860/jobs/6209513237)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/3672847860/jobs/6209513237","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`dist-riscv64-linux`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/3672847860/jobs/6209394684) [(plain)](https://github.com/rust-lang-ci/rust/commit/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/checks/10027840463/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling std_detect v0.1.5 (/checkout/library/stdarch/crates/std_detect)
[RUSTC-TIMING] rustc_std_workspace_alloc test:false 0.024
[RUSTC-TIMING] panic_abort test:false 0.045
[RUSTC-TIMING] panic_unwind test:false 0.076
error: internal compiler error: no warnings or errors encountered even though `delayed_good_path_bugs` issued

error: internal compiler error: trimmed_def_paths constructed
  |
  = note: delayed at    0: <rustc_errors::Handler>::delay_good_path_bug::<&str>
             1: rustc_middle::ty::print::pretty::trimmed_def_paths
             2: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::ArenaCache<(), std::collections::hash::map::HashMap<rustc_span::def_id::DefId, rustc_span::symbol::Symbol, core::hash::BuildHasherDefault<rustc_hash::FxHasher>>>>
             3: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::trimmed_def_paths, rustc_query_impl::plumbing::QueryCtxt>
             4: <rustc_middle::ty::print::pretty::FmtPrinter as rustc_middle::ty::print::Printer>::print_def_path
             5: <rustc_middle::ty::context::TyCtxt>::def_path_str
             6: <rustc_hir::hir::Pat>::walk_::<<rustc_hir::hir::Pat>::walk_always<rustc_mir_build::thir::pattern::check_match::check_for_bindings_named_same_as_variants::{closure#0}>::{closure#0}>
             7: <rustc_mir_build::thir::pattern::check_match::MatchVisitor>::check_irrefutable
             8: <rustc_mir_build::thir::pattern::check_match::MatchVisitor as rustc_hir::intravisit::Visitor>::visit_local
             9: rustc_hir::intravisit::walk_expr::<rustc_mir_build::thir::pattern::check_match::MatchVisitor>
            10: <rustc_mir_build::thir::pattern::check_match::MatchVisitor as rustc_hir::intravisit::Visitor>::visit_expr
            11: rustc_mir_build::thir::pattern::check_match::check_match
            12: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, ()>>
            13: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::check_match, rustc_query_impl::plumbing::QueryCtxt>
            14: rustc_data_structures::sync::par_for_each_in::<&[rustc_span::def_id::LocalDefId], <rustc_middle::hir::map::Map>::par_body_owners<rustc_interface::passes::analysis::{closure#1}::{closure#0}::{closure#0}::{closure#0}>::{closure#0}>
            15: <rustc_session::session::Session>::time::<(), rustc_interface::passes::analysis::{closure#1}::{closure#0}::{closure#0}>
            16: <rustc_session::session::Session>::time::<(), rustc_interface::passes::analysis::{closure#1}>
            18: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::plumbing::QueryCtxt, rustc_query_system::query::caches::DefaultCache<(), core::result::Result<(), rustc_errors::ErrorGuaranteed>>>
            19: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::analysis, rustc_query_impl::plumbing::QueryCtxt>
            20: <rustc_interface::passes::QueryContext>::enter::<rustc_driver::run_compiler::{closure#1}::{closure#2}::{closure#2}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
            21: <rustc_interface::interface::Compiler>::enter::<rustc_driver::run_compiler::{closure#1}::{closure#2}, core::result::Result<core::option::Option<rustc_interface::queries::Linker>, rustc_errors::ErrorGuaranteed>>
            21: <rustc_interface::interface::Compiler>::enter::<rustc_driver::run_compiler::{closure#1}::{closure#2}, core::result::Result<core::option::Option<rustc_interface::queries::Linker>, rustc_errors::ErrorGuaranteed>>
            22: rustc_span::with_source_map::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_interface::interface::run_compiler<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
            23: <scoped_tls::ScopedKey<rustc_span::SessionGlobals>>::set::<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
            24: std::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>::{closure#0}::{closure#0}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
            25: <<std::thread::Builder>::spawn_unchecked_<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>::{closure#0}::{closure#0}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
            26: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
                       at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/alloc/src/boxed.rs:1990:9
            27: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
                       at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/alloc/src/boxed.rs:1990:9
                       at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/sys/unix/thread.rs:108:17
            29: start_thread
            30: clone
          
          

thread 'rustc' panicked at 'Box<dyn Any>', compiler/rustc_errors/src/lib.rs:1610:13
stack backtrace:
   0:     0x7f19763da010 - std::backtrace_rs::backtrace::libunwind::trace::h2540c8c07dbc0a17
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f19763da010 - std::sys_common::backtrace::_print_fmt::h6451477ed0f776b0
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x7f19763da010 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h576e21ae7c0954a4
   3:     0x7f19763da010 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h576e21ae7c0954a4
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f1976439fbe - core::fmt::write::hcb131ddd0c15f718
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/core/src/fmt/mod.rs:1208:17
   5:     0x7f19763cbd05 - std::io::Write::write_fmt::hec164e5640917b03
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/io/mod.rs:1682:15
   6:     0x7f19763d9dd5 - std::sys_common::backtrace::_print::h6a73fa5166776a40
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x7f19763d9dd5 - std::sys_common::backtrace::print::h57ad279d5588234c
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x7f19763dcb2f - std::panicking::default_hook::{{closure}}::hc4b6fa6506eb52c2
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/panicking.rs:267:22
   9:     0x7f19763dc86a - std::panicking::default_hook::h26b88df41b3e344d
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/panicking.rs:286:9
  10:     0x7f197713b246 - rustc_driver[c7c666aeec07f119]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x7f19763dd2f9 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h29104a2a0427fe95
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/alloc/src/boxed.rs:2024:9
  12:     0x7f19763dd2f9 - std::panicking::rust_panic_with_hook::h81de260a74888be4
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/panicking.rs:692:13
  13:     0x7f197ccd0573 - std[c962dfd4bdbe76a2]::panicking::begin_panic::<rustc_errors[b92b891ac752ae31]::ExplicitBug>::{closure#0}
  14:     0x7f197ccccc36 - std[c962dfd4bdbe76a2]::sys_common::backtrace::__rust_end_short_backtrace::<std[c962dfd4bdbe76a2]::panicking::begin_panic<rustc_errors[b92b891ac752ae31]::ExplicitBug>::{closure#0}, !>
  15:     0x7f1976fd2c46 - std[c962dfd4bdbe76a2]::panicking::begin_panic::<rustc_errors[b92b891ac752ae31]::ExplicitBug>
  16:     0x7f197ccc2956 - std[c962dfd4bdbe76a2]::panic::panic_any::<rustc_errors[b92b891ac752ae31]::ExplicitBug>
  17:     0x7f197ccc5ec8 - <rustc_errors[b92b891ac752ae31]::HandlerInner as core[ae65f184c67bc919]::ops::drop::Drop>::drop
  18:     0x7f19770b9888 - core[ae65f184c67bc919]::ptr::drop_in_place::<rustc_session[31f131ebab3b1e65]::parse::ParseSess>
  19:     0x7f19770bb128 - core[ae65f184c67bc919]::ptr::drop_in_place::<rustc_session[31f131ebab3b1e65]::session::Session>
  20:     0x7f19770a2e59 - core[ae65f184c67bc919]::ptr::drop_in_place::<rustc_interface[77ad0e590bfcca6b]::interface::Compiler>
  21:     0x7f197709fa24 - rustc_span[b384040717ceefb]::with_source_map::<core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>, rustc_interface[77ad0e590bfcca6b]::interface::run_compiler<core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>, rustc_driver[c7c666aeec07f119]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  22:     0x7f1977102a08 - <scoped_tls[27cd29d9afbae785]::ScopedKey<rustc_span[b384040717ceefb]::SessionGlobals>>::set::<rustc_interface[77ad0e590bfcca6b]::interface::run_compiler<core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>, rustc_driver[c7c666aeec07f119]::run_compiler::{closure#1}>::{closure#0}, core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>>
  23:     0x7f19770c9880 - std[c962dfd4bdbe76a2]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[77ad0e590bfcca6b]::util::run_in_thread_pool_with_globals<rustc_interface[77ad0e590bfcca6b]::interface::run_compiler<core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>, rustc_driver[c7c666aeec07f119]::run_compiler::{closure#1}>::{closure#0}, core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>>
  24:     0x7f19770aaa04 - <<std[c962dfd4bdbe76a2]::thread::Builder>::spawn_unchecked_<rustc_interface[77ad0e590bfcca6b]::util::run_in_thread_pool_with_globals<rustc_interface[77ad0e590bfcca6b]::interface::run_compiler<core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>, rustc_driver[c7c666aeec07f119]::run_compiler::{closure#1}>::{closure#0}, core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ae65f184c67bc919]::result::Result<(), rustc_errors[b92b891ac752ae31]::ErrorGuaranteed>>::{closure#1} as core[ae65f184c67bc919]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  25:     0x7f19763e70f3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h85a72ba036d4c02b
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/alloc/src/boxed.rs:1990:9
  26:     0x7f19763e70f3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h6b17c7b56a9554b9
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/alloc/src/boxed.rs:1990:9
  27:     0x7f19763e70f3 - std::sys::unix::thread::Thread::new::thread_start::h4e013c1a945d5337
                               at /rustc/3ece57ecf4c9f308fe935a733028dd53e5b2c69c/library/std/src/sys/unix/thread.rs:108:17
  28:     0x7f197611a609 - start_thread
  29:     0x7f197625a133 - clone
  30:                0x0 - <unknown>
note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md


note: rustc 1.68.0-nightly (3ece57ecf 2022-12-12) running on x86_64-unknown-linux-gnu

note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 -C debuginfo=1 -Z unstable-options -C linker=riscv64-unknown-linux-gnu-gcc -C symbol-mangling-version=legacy -Z unstable-options -Z unstable-options -Z macro-backtrace -C link-args=-Wl,-z,origin -C link-args=-Wl,-rpath,$ORIGIN/../lib -C split-debuginfo=off -Z save-analysis -C prefer-dynamic -C embed-bitcode=yes -C force-unwind-tables=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -Z force-unstable-if-unmarked
note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
```

</details>

Commit:
Remove (eager)

Commit:
Try to fix ICE

mejrs commented:
I think this fixes the ice that was hit during rollup, but I'm not sure how to test it locally.

mejrs commented:
I think this works on `dist-riscv64-linux` now - see https://github.com/rust-lang/rust/actions/runs/3721937031/jobs/6312424723

Nilstrieb commented:
@bors r=davidtwco
Let's hope it really works now

bors commented:
:pushpin: Commit f7e894c000d6526c32422c6282ce277c05499a94 has been approved by `davidtwco`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=davidtwco f7e894c000d6526c32422c6282ce277c05499a94 -->
<!-- homu: {"type":"Approved","sha":"f7e894c000d6526c32422c6282ce277c05499a94","approver":"davidtwco","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit f7e894c000d6526c32422c6282ce277c05499a94 with merge 35a99eef32a2b7b9d8e77dde539f869e522d181f...
<!-- homu: {"type":"BuildStarted","head_sha":"f7e894c000d6526c32422c6282ce277c05499a94","merge_sha":"35a99eef32a2b7b9d8e77dde539f869e522d181f"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3724087657/jobs/6316967369)
Approved by: davidtwco
Pushing 35a99eef32a2b7b9d8e77dde539f869e522d181f to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"davidtwco","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/3724087657/jobs/6316967369"},"merge_sha":"35a99eef32a2b7b9d8e77dde539f869e522d181f"} -->

rust-timer commented:
Finished benchmarking commit (35a99eef32a2b7b9d8e77dde539f869e522d181f): [comparison URL](https://perf.rust-lang.org/compare.html?start=48b3c4612681ba7828880aa9e675452c62714bbf&end=35a99eef32a2b7b9d8e77dde539f869e522d181f&stat=instructions:u).

### Overall result: ❌✅ regressions and improvements - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=48b3c4612681ba7828880aa9e675452c62714bbf&end=35a99eef32a2b7b9d8e77dde539f869e522d181f&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | 1.0%  | [1.0%, 1.0%]   | 1     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -1.3% | [-1.3%, -1.3%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=48b3c4612681ba7828880aa9e675452c62714bbf&end=35a99eef32a2b7b9d8e77dde539f869e522d181f&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 2.3% | [2.3%, 2.3%] | 1     |
| Regressions ❌ <br /> (secondary)  | 1.3% | [1.3%, 1.3%] | 2     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | 2.3% | [2.3%, 2.3%] | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=48b3c4612681ba7828880aa9e675452c62714bbf&end=35a99eef32a2b7b9d8e77dde539f869e522d181f&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.
